### PR TITLE
[FEAT] Route widget on click to image (#1418)

### PR DESCRIPTION
**Pull request by prateekmedia**
## Description

Previously clicking on it was opening favorites album but now it will open favorites album as well as image.

### CODE REVIEW

**Comment by prateekmedia** [comment]
Merge this to main when deps PR is resolved as this depends on that.

File: mobile/lib/app.dart [hunk 1]
```diff
@@ -89,17 +91,38 @@ class _EnteAppState extends State<EnteApp> with WidgetsBindingObserver {
...
+
+    final previousGeneratedId =
+        await hw.HomeWidget.getWidgetData<int>("home_widget_last_img");
+    final res = previousGeneratedId != null
+        ? await FilesDB.instance.getFile(
+            previousGeneratedId,
+          )
+        : null;
+
+    routeToPage(
```
Comment (vishnukvmd) [inline mobile/lib/app.dart]: Curious, isn't the linter complaining because there's a missing `await`?

Comment (prateekmedia) [reply mobile/lib/app.dart]: See the `.ignore()` at the end.

File: mobile/lib/app.dart [hunk 2]
```diff
@@ -89,17 +91,38 @@ class _EnteAppState extends State<EnteApp> with WidgetsBindingObserver {
...
+      context,
+      CollectionPage(
+        CollectionWithThumbnail(
+          collection,
+          thumbnail,
         ),
       ),
+    ).ignore();
+
+    if (previousGeneratedId == null) return;
```
Comment (vishnukvmd) [inline mobile/lib/app.dart]: Is this check necessary? Doesn't the next check take care of this as well?

**Review by vishnukvmd — COMMENTED** [review]
@prateekmedia LGTM! Will merge once it's out of Drafts.

File: mobile/lib/app.dart (hunk 2 above)
Comment (prateekmedia) [reply mobile/lib/app.dart]: You're right.

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**